ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterAggregator.java

```diff
@@ -116,13 +139,47 @@ private void combineRaw(StatsSnapshot rawBaseSnapshot, StatsWrapper snapshotWrap
    *       given partition entry has a greater value.
    *    b) Timestamp of a given partition entry is one aggregation period newer (greater) than the timestamp of the
    *       base entry.
+   * The combined valid stats snapshot is represented in following format:
+   * {
+   *   value: 1000,
+   *   subMap: {
+   *     Partition[0]: {
+   *       value: 400,
+   *       subMap: {
+   *         Account[0]: {
+   *           value: 400,
+   *           subMap: {
+   *             Container[0]: {
+   *               value: 400
+   *               subMap: null
+   *             }
+   *           }
+   *         }
+   *       }
+   *     },
+   *     Partition[1]: {
+   *       value: 600,
+   *       subMap: {
+   *         Account[1]: {
+   *           value: 600,
+   *           subMap: {
+   *             Container[1]:{
+   *               value: 600,
+   *               subMap: null
+   *             }
+   *           }
+   *         }
+   *       }
+   *     }
+   *   }
+   * }
    * @param baseSnapshot the base {@link StatsSnapshot} which will contain the aggregated result
    * @param snapshotWrapper the {@link StatsSnapshot} to be aggregated to the base {@link StatsSnapshot}
    * @param instance new instance from which snapshot is being combined
    * @param partitionTimestampMap a {@link Map} of partition to timestamp to keep track the current timestamp of each
    *                              partition entry in the base {@link StatsSnapshot}
    */
-  private void combine(StatsSnapshot baseSnapshot, StatsWrapper snapshotWrapper, String instance,
+  private void combineValidStatsByAccount(StatsSnapshot baseSnapshot, StatsWrapper snapshotWrapper, String instance,
```

Comment:
the two `combineValidStatsBy*` methods appear to have a lot of duplicated code. It might be possible to combine some of it, but I see that there are some subtle differences between the two that might make it difficult. Might be worth looking into in a future PR, though.

Reply:
Thanks for pointing this out. I think I would try to combine those two methods in the future.